### PR TITLE
Add wallet balance option in the sidedrawer #41

### PR DESCRIPTION
**Pull request by chandra-pro**
Added the wallet component in sidebar. As of now Apollo is showing error of network request failed so i comment out that code.

### CODE REVIEW

File: components/messages/Message.tsx [hunk 1]
```diff
@@ -20,18 +19,22 @@ const Message = ({ time, isLeft, message, onSwipe }: any) => {
...
+    } else if (!isLeft && type === 'time') {
       return {
-        color: 'darkgray',
+        color: 'black',
       };
     } else if (isLeft && type === 'message') {
       return {
         color: 'white',
       };
+    } else if (!isLeft && type === 'message') {
```
Comment (mdshamoon) [inline components/messages/Message.tsx:34]: Please update this function with a switch case:

const isOnLeft = (type: any) => {
  switch (type) {
    case 'time':
    case 'message':
      return isLeft ? { color: 'white' } : { color: 'black' };

    case 'messageContainer':
      return isLeft
        ? {
            alignSelf: 'flex-start',
            backgroundColor: '#073F24',
            borderTopLeftRadius: 0,
          }
        : {
            borderTopRightRadius: 0,
            color: '#E1E8E5',
          };

    default:
      return {
        borderTopRightRadius: 0,
        color: '#E1E8E5',
      };
  }
};

**Review by mdshamoon — CHANGES_REQUESTED** [review]
Let me know what is the issue you are getting while calling the walletBallance API

**Comment by chandra-pro** [comment]
> Let me know what is the issue you are getting while calling the walletBallance API

I was getting error of network request failed.

**Comment by mdshamoon** [comment]
@chandra-pro it can only be called under the apollo client provider. make sure your component is a child of the apollo client provider

**Comment by chandra-pro** [comment]
@mdshamoon I have done what you have told but data is not rendered. Please look into the code

**Comment by mdshamoon** [comment]
@chandra-pro I meant the ApolloProvider we already defined in Chat.tsx file

The way it works is when you make a call with useQuery it checks if it needs to add any header or other configurations from the ApolloProvider anywhere in the parent chain. We have added auth header in the ApolloProvider used in Chat.tsx

The way you can do is to move ApolloProvider out of Chat.tsx and add in App.tsx 
With this Wallet component will be a child of ApolloProvider and the call will be made with the authentication header.

Let me know if this makes sense. Or we can also have a call and I can explain

**Comment by chandra-pro** [comment]
I have already use Apollo provider in app.tsx to make the component as a
child of this.
Please see the code once.

**Comment by mdshamoon** [comment]
You have created a separate client and not used the one already present which is configured for adding header.

image

Please see if that works and remove from the provider from Chat.tsx once you add here.

File: components/ui/Wallet.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    <View style={styles.walletcontainer}>
+      <View style={styles.innercontainer}>
+        <MaterialCommunityIcons
+          name="wallet-outline"
+          size={24}
+          color="white"
+          style={styles.walleticon}
+        />
+        <Text style={styles.walletText}>Your Wallet Balance</Text>
+        <View>
```
Comment (mdshamoon) [inline components/ui/Wallet.tsx:24]: @chandra-pro I have checked the code by pulling on local and I can see that you are getting the data as an object. 

add a console here and you will see that you are getting the data in this format {"bspbalance": "{\"balance\":4.874}"} which is an object.

You have to extract the string from here and then display it

**Comment by mdshamoon** [comment]
@chandra-pro want to have a quick call if available to check the issue

**Comment by chandra-pro** [comment]
yeah available

**Comment by mdshamoon** [comment]
@chandra-pro are you on discord? If yes can you share your name or id I can send the link there

**Comment by chandra-pro** [comment]
aldehyde#1074

**Comment by chandra-pro** [comment]
@mdshamoon still getting Apollo Error and the value of data as undefined.
Could you please help me regarding this.

**Comment by mdshamoon** [comment]
@chandra-pro let me take over this since it is working fine for me. Can you complete #52 meanwhile

**Comment by mdshamoon** [comment]
Closing in favor of #65

**Pull request closed without merging**